### ISSUE TITLE
HtmlEncoded fullwidth attacks are not being detected

### ISSUE THREAD

**Issue by xzhang-ipipeline** [coreruleset/coreruleset#3834]
### Describe the bug

Fullwidth attacks can be detected, for example: `＜ｓｃｒｉｐｔ＞ａｌｅｒｔ（＂ｅｎｃｏｄｅｄ＂）＜／ｓｃｒｉｐｔ＞`
However, when it's submitted through a form, the browser encode it and then it's not being detected, for example:
`&#65308;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;&#65345;&#65356;&#65349;&#65362;&#65364;&#65288;&#65282;&#65349;&#65358;&#65347;&#65359;&#65348;&#65349;&#65348;&#65282;&#65289;&#65308;&#65295;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;
`
### Steps to reproduce
curl -X POST -i --data '&#65308;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;&#65345;&#65356;&#65349;&#65362;&#65364;&#65288;&#65282;&#65349;&#65358;&#65347;&#65359;&#65348;&#65349;&#65348;&#65282;&#65289;&#65308;&#65295;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;' https://your.target.site

### Expected behaviour

Expect below rules are being detected:
941100
941110
941160
941320

### Actual behaviour

None of above rules are triggered

### Additional context
Currently the 941 rules have transformations in order: `t:none,t:utf8toUnicode,t:urlDecodeUni,t:htmlEntityDecode,t:jsDecode,t:cssDecode,t:removeNulls`

However, to handle this case, there needs to be t:utf8toUnicode after t:htmlEntityDecode

### Your Environment
Kemp LoadMaster

* CRS version (e.g., v3.2.0): 3.3.4
* Paranoia level setting: 4
* ModSecurity version (e.g., 2.9.3): 2.9.6
* Web Server and version (e.g., apache 2.4.41): Kemp LoadMaster
* Operating System and version: Kemp LoadMaster

**Comment by xzhang-ipipeline** [comment]
Sorry, the reproduce code got decoded by github. Let me try again.


curl -X POST -i --data '&#65308;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;&#65345;&#65356;&#65349;&#65362;&#65364;&#65288;&#65282;&#65349;&#65358;&#65347;&#65359;&#65348;&#65349;&#65348;&#65282;&#65289;&#65308;&#65295;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;' https://your.target.site

**Comment by azurit** [comment]
@xzhang-ipipeline Hi, this is detected on PL1 by CRS4:

```
$ curl -X POST -i --data '＜ｓｃｒｉｐｔ＞ａｌｅｒｔ（＂ｅｎｃｏｄｅｄ＂）＜／ｓｃｒｉｐｔ＞' -H "x-format-output: txt-matched-rules" https://sandbox.coreruleset.org/
HTTP/1.1 403 Forbidden
Date: Fri, 27 Sep 2024 10:11:34 GMT
Content-Type: text/plain; charset=iso-8859-1
Transfer-Encoding: chunked
Connection: keep-alive
X-Unique-ID: ZvaE1vtqkgXpmHEzBFWkKwAAABE
x-backend: apache-nightly
x-crs-last-commit: none

941100 PL1 XSS Attack Detected via libinjection
941110 PL1 XSS Filter - Category 1: Script Tag Vector
941160 PL1 NoScript XSS InjectionChecker: HTML Injection
949110 PL1 Inbound Anomaly Score Exceeded (Total Score: 15)
980170 PL1 Anomaly Scores: (Inbound Scores: blocking=15, detection=15, per_pl=15-0-0-0, threshold=5) - (Outbound Scores: blocking=0, detection=0, per_pl=0-0-0-0, threshold=4) - (SQLI=0, XSS=15, RFI=0, LFI=0, RCE=0, PHPI=0, HTTP=0, SESS=0, COMBINED_SCORE=15)
```

**Comment by xzhang-ipipeline** [comment]
Thanks @azurit  for getting back to me. What you tested was the non-htmlencoded fullwidth. This is what I got using below:

`curl -X POST -i --data '&#65308;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;&#65345;&#65356;&#65349;&#65362;&#65364;&#65288;&#65282;&#65349;&#65358;&#65347;&#65359;&#65348;&#65349;&#65348;&#65282;&#65289;&#65308;&#65295;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;' -H "x-format-output: txt-matched-rules" https://sandbox.coreruleset.org/`

![image](https://github.com/user-attachments/assets/f0a2ec9b-c34d-4a93-9ca1-2b56f3c52747)

**Comment by RedXanadu** [comment]
Please excuse my ignorance, but are there applications that are actually decoding and executing input like
`&#65308;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;&#65345;`
?

Is this a valid attack vector / do you have an example you can share?

**Comment by xzhang-ipipeline** [comment]
What happens is that when `＜ｓｃｒｉｐｔ＞ａｌｅｒｔ（＂ｅｎｃｏｄｅｄ＂）＜／ｓｃｒｉｐｔ＞` is entered into a form field, the browser will encode it to `&#65308;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;&#65345;&#65356;&#65349;&#65362;&#65364;&#65288;&#65282;&#65349;&#65358;&#65347;&#65359;&#65348;&#65349;&#65348;&#65282;&#65289;&#65308;&#65295;&#65363;&#65347;&#65362;&#65353;&#65360;&#65364;&#65310;`  to post back to the server. 

Does this help?

**Comment by xzhang-ipipeline** [comment]
This is what the browser sent after I entered `＜ｓｃｒｉｐｔ＞ａｌｅｒｔ（＂ｅｎｃｏｄｅｄ＂）＜／ｓｃｒｉｐｔ＞` in the form field "txtDescription".
![image](https://github.com/user-attachments/assets/2b725363-cb3b-4c5d-ab86-08fd0e51e765)
